### PR TITLE
Show channels in the depot UI

### PR DESCRIPTION
**Pull request by cnunciato**
Adds channels to the package-list and package-detail views using the new `/channels` API.

Signed-off-by: Christian Nunciato <cnunciato@chef.io>

### CODE REVIEW

**Comment by thesentinels** [comment]
Thanks for the pull request! Here is what will happen next:
 1. Your PR will be reviewed by the maintainers
 2. If everything looks good, one of them will `approve` it, and your PR will be merged.

Thank you for contributing!

**Comment by cnunciato** [comment]
Resolves #2646.

File: components/builder-web/app/actions/packages.ts [hunk 1]
```diff
@@ -68,12 +78,38 @@ export function fetchPackage(pkg) {
...
+            if (pkg.origin && pkg.name && pkg.version && pkg.release) {
+                depotApi.getPackageChannels(pkg)
+                    .then(channels => {
+                        dispatch(setVisiblePackageChannels(pkg, channels));
+                    })
+                .catch(error => console.error(error));
+            }
+        });
+    };
+}
```
Comment (raskchanky) [inline components/builder-web/app/actions/packages.ts]: Using the existing APIs, this is definitely a good way to accomplish the goal of retrieving a list of channels for every package.  I'm not opposed to this at all.  I do think it _might_ be possible to accomplish your goal without having the overhead of making an HTTP request for every package.  I'm going to try and spike out a quick POC for what I mean.

**Comment by chefsalim** [comment]
Nice work!  Looks great to me.

tenor-67357025

**Comment by chefsalim** [comment]
@thesentinels approve

**Comment by thesentinels** [comment]
:metal: I am testing your branch against master before merging it. We do this to ensure that the master branch is never failing tests.

**Comment by thesentinels** [comment]
:neckbeard: Travis CI has [started testing this PR](https://travis-ci.org/habitat-sh/habitat/builds/252625293).

**Comment by thesentinels** [comment]
:sparkling_heart: Travis CI [reports this PR passed](https://travis-ci.org/habitat-sh/habitat/builds/252625293).

It always makes me feel nice when humans approve of one anothers work. I'm merging this PR now.

I just want you and the contributor to answer me one question:

gif-keyboard-3280869874741411265

File: components/builder-web/app/actions/packages.ts (hunk 1 above)
Comment (raskchanky) [inline components/builder-web/app/actions/packages.ts]: Here is a quick POC that I did just now to show what I was talking about on your PR, as an alternative to making N+1 HTTP requests to fetch channels for packages.  It's pretty rough, and only implemented for listing packages, but I think it at least shows what's possible.  https://github.com/habitat-sh/habitat/compare/jb/lulz?expand=1

Comment (raskchanky) [inline components/builder-web/app/actions/packages.ts]: And this is what the end result looks like:

**Pull request merged**